build/pip/setup.py

```diff
@@ -0,0 +1,122 @@
+"""The setup installs Gauge CLI through pip."""
+from setuptools import setup
+from os import path
+from setuptools.command.install import install
+import json
+import platform
+import glob
+import zipfile
+import shutil
+import os
+import sys
+try:
+    import requests
+
+except ImportError:
+    os.system('python -m ensurepip --default-pip')
+    os.system('python -m pip install requests')
+    import requests
+
+# TODO: bumped manually for now, wish to be automatically picked from source
+release_ver = '1.0.2'
```

Comment:
I might need some help/info around this , whether to create a new file or suggest any specific file from which I can pick `environment` variable

Reply:
There no specific file for the `version`. It can pick from process environment variables.
In examples : 
`version=os.environ['GAUGE_VERSION']`

We can set this env while building the package on CI.

Reply:
The Only reason for the first merge to be static is pretty simple.
`pip install gauge-cli --version=1.0.0` will not download or install an older version as the package is not published to PyPi yet.
This is the actual plan of publish it with the necessary executables and then move it to be dynamic so that the push will happen along with CI, this would take some time to resolve in case if it needs to be dynamic from the start

Reply:
Ok, Get it.

Reply:
We probably have to add a script which generates the `setup.py` dynamically based on the `VERSION`. It's been done in our `gauge-pytho` project.

Reference :- https://github.com/getgauge/gauge-python/blob/3f0618a9492dc67c7474b326d5b4db33f9b7995d/build.py#L34 and https://github.com/getgauge/gauge-python/blob/master/setup.tmpl